core/src/main/java/com/netflix/conductor/service/ExecutionService.java

```diff
@@ -174,20 +175,18 @@ public Task getPendingTaskForWorkflow(String taskReferenceName, String workflowI
 		return executor.getPendingTaskByWorkflow(taskReferenceName, workflowId);
 	}
 
-	public boolean ackTaskRecieved(String taskId, String consumerId) throws Exception {
-		Task task = getTask(taskId);
-
-		if (task != null) {
-			String queueName = QueueUtils.getQueueName(task);
-			if(task.getResponseTimeoutSeconds() > 0) {
-				logger.debug("Adding task " + queueName + "/" + taskId + " to be requeued if no response received " + task.getResponseTimeoutSeconds());
-				return queue.setUnackTimeout(queueName, task.getTaskId(), 1000 * task.getResponseTimeoutSeconds());		//Value is in millisecond
-			}else {
-				return queue.ack(queueName, taskId);
-			}
-		}
-		return false;
-
+	/**
+	 * This method removes the task from the un-acked Queue
+	 *
+	 * @param taskId: the taskId that needs to be updated and removed from the unacked queue
+	 * @throws Exception In case of an error while getting the Task from the executionDao
+	 * @return: True in case of successful removal of the taskId from the un-acked queue
+	 */
+	public boolean ackTaskReceived(String taskId) throws Exception {
```

Comment:
It would be useful to log the `workerid` that sends the ack here, unless this is logged somewhere else.

Reply:
Good point.

Reply:
The TaskResource endpoint has not changed and still accepts the consumerId as a query param. 

```
@POST
	@Path("/{taskId}/ack")
	@ApiOperation("Ack Task is recieved")
	@Consumes({ MediaType.WILDCARD })
	public String ack(@PathParam("taskId") String taskId, @QueryParam("workerid") String workerId) throws Exception {
		return "" + taskService.ackTaskReceived(taskId);
	}
```

The access logs will still show the task being acked and also the consumer id being used for the ack.